src/app/subnets/views/VLANDetails/VLANDetails.tsx

```diff
@@ -91,13 +91,10 @@ const VLANDetails = (): JSX.Element => {
           <VLANSummary id={id} />
           <DHCPStatus id={id} openForm={() => setShowDHCPForm(true)} />
           <ReservedRanges hasVLANSubnets={subnets.length > 0} vlanId={id} />
-          <VLANSubnets id={id} />
+          <VLANSubnets subnets={subnets} />
```

Comment:
It would be better to remain consistent with other components at this level which take vlan id as a prop and select necessary state slice internally. You can look at `src/app/subnets/views/VLANDetails/DHCPStatus/DHCPStatus.tsx` for an example.

```suggestion
          <VLANSubnets id={id} />
```

Reply:
To save us some hassle you _could_ combine the two selectors below into a React hook that takes the VLAN id and returns a list of subnets.

```tsx
  const vlan = useSelector((state: RootState) =>
    vlanSelectors.getById(state, id)
  );
  const vlanSubnets = useSelector((state: RootState) =>
    subnetSelectors.getByIds(state, vlan ? vlan.subnet_ids : [])
  );
  ```